src/librustc_typeck/check/pat.rs

```diff
@@ -97,11 +97,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 self.check_pat_struct(pat, qpath, fields, *etc, expected, def_bm, discrim_span)
             }
             PatKind::Or(pats) => {
-                let expected_ty = self.structurally_resolved_type(pat.span, expected);
```

Comment:
@dlrobertson Was there a specific reason for `.structurally_resolved_type` here?